src/ssd.c

```diff
@@ -423,10 +423,22 @@ ssd_part_contains(enum ssd_part_type whole, enum ssd_part_type candidate)
 		return true;
 	}
 	if (whole == LAB_SSD_PART_TITLEBAR) {
-		return candidate >= LAB_SSD_BUTTON_CLOSE && candidate <= LAB_SSD_PART_CORNER_TOP_RIGHT;
+		return candidate >= LAB_SSD_BUTTON_CLOSE && candidate <= LAB_SSD_PART_TITLE;
 	}
 	if (whole == LAB_SSD_FRAME) {
 		return candidate >= LAB_SSD_BUTTON_CLOSE && candidate <= LAB_SSD_CLIENT;
 	}
+	if (whole == LAB_SSD_PART_TOP) {
```

Comment:
Is there a cleaner implementation that avoids using sdd_part_contains()?

What is the situation that you can't handle by just iterating over them and get to the right one? When the parts are created, they are carefully ordered so that rendering / pressing should just be able to iterate over the list and stop at the first match.